[GPU][DT] Expand inner dimensions during materialization.

hanhanW opened the pull request:
Previously we used "innerTiles[0]xinnerTiles[1]" layout for the materialized shape; treat it like opaque shape, and it relies on later lowering to resolve the opaque type. However, it is quite tricky to represent the core contraction op with multi_mma op. This revisions removes those dummy reshapes, and generates the target shape directly.

The revision also removes unnecessary methods from GPUMaterializeEncoding.cpp, since the MaterializeEncodingInfo carries the information in the patch.

Max191 commented on compiler/src/iree/compiler/Codegen/Common/GPU/GPUMaterializeEncoding.cpp:
```diff
@@ -257,26 +250,21 @@ struct GPUSetEncodingOpLoweringConversion
-      targetShape = {intrinsicShape->M, intrinsicShape->K};
+      targetShape = {maybeEncodingInfo->intrinsicSize[0],
+                     maybeEncodingInfo->intrinsicSize[2]};
       break;
     case 1: // B
-      targetShape = {intrinsicShape->N, intrinsicShape->K};
+      targetShape = {maybeEncodingInfo->intrinsicSize[1],
+                     maybeEncodingInfo->intrinsicSize[2]};
       break;
     case 2: // C
-      targetShape = {intrinsicShape->M, intrinsicShape->N};
+      targetShape = {maybeEncodingInfo->intrinsicSize[0],
+                     maybeEncodingInfo->intrinsicSize[1]};
       break;
```
What is the purpose of `targetShape`/`intrinsicSize`? I thought everything we need for the swizzling was in `innerTileSizes`, `innerTileShapes`, `permutation`.

hanhanW replied:
I initiated it for unrolling. Basically `targetShape/intrinsicSize` is the unrolling factor. @lialan is going to add such support for the materialization part.

Max191 reviewed (approved):
LGTM, thanks Hanhan!

lialan reviewed (approved):
Thanks, this infra will make adding tiling and unrolling easier.

hanhanW committed:
[GPU][DT] Expand inner dimensions during materialization.

Previously we used "innerTiles[0]xinnerTiles[1]" layout for the
materialized shape; treat it like opaque shape, and it relies on later
lowering to resolve the opaque type. However, it is quite tricky to
represent the core contraction op with multi_mma op. This revisions
removes the those dummy reshapes, and generates the target shape
directly.

The revision also removes unnecessary methods from
GPUMaterializeEncoding.cpp, since the MaterializeEncodingInfo carries
the information in the patch.

Signed-off-by: hanhanW <hanhan0912@gmail.com>
